en/12_Ecosystem_Utilities_and_Compatibility.adoc

```diff
@@ -0,0 +1,542 @@
+
+== Introduction
+
+In this chapter, we'll explore important ecosystem utilities for Vulkan development and learn how to adapt our code to support a wider range of GPUs. As Vulkan continues to evolve with new versions and features, it's important to understand how to:
+
+1. Discover what features are supported by different GPUs
+2. Modify your code to maintain compatibility with older hardware
+3. Conditionally use advanced features when available
+
+This knowledge is essential for developing Vulkan applications that can run on a diverse range of hardware, from the latest high-end GPUs to older or more limited devices.
+
+== Vulkan Hardware Database (GPUInfo.org)
+
+=== Introduction to GPUInfo.org
+
+The link:https://vulkan.gpuinfo.org/[Vulkan Hardware Database] (GPUInfo.org) is an invaluable resource for Vulkan developers. This community-driven database collects and presents information about Vulkan support across a wide range of GPUs and devices.
+
+GPUInfo.org provides detailed information about:
+
+* Supported Vulkan versions
+* Available extensions
+* Feature support
+* Implementation limits
+* Format properties
+* Queue family properties
+
+This information is crowdsourced from users who run the Vulkan Hardware Capability Viewer tool, which reports their GPU's capabilities to the database.
+
+=== Using GPUInfo.org for Development
+
+When developing a Vulkan application, GPUInfo.org can help you:
+
+1. *Determine minimum requirements*: Understand what Vulkan version and extensions you need to target to support your desired range of hardware.
+
+2. *Check feature availability*: Verify if specific features like dynamic rendering, timeline semaphores, or ray tracing are widely supported.
+
+3. *Identify implementation limits*: Discover the practical limits of various Vulkan features across different hardware.
+
+4. *Compare vendors and devices*: Understand the differences in Vulkan support between NVIDIA, AMD, Intel, and mobile GPU vendors.
+
+Let's look at some practical examples of using GPUInfo.org:
+
+==== Example: Checking Vulkan Version Support
+
+To determine how widely supported Vulkan 1.3 (which introduced dynamic rendering) is:
+
+1. Visit link:https://vulkan.gpuinfo.org/[GPUInfo.org]
+2. Navigate to "Core Version Support"
+3. Check the percentage of devices supporting Vulkan 1.3
+
+You'll find that while newer GPUs support Vulkan 1.3+, there are still many devices limited to Vulkan 1.0, 1.1, or 1.2.
+
+==== Example: Checking Extension Support
+
+If you're considering using a specific extension:
+
+1. Visit the "Extensions" section
+2. Search for your extension of interest
+3. Check its support percentage across different vendors
+
+This helps you decide whether to require the extension or provide a fallback path.
+
+==== Example: Using the vulkanconfig Tool
+
+The `vulkanconfig` tool is included in the Vulkan SDK and provides a convenient way to configure Vulkan settings on your system. Here's how to use it:
```

Comment:
On windows it's 'vkconfig' (.exe). But running that from the command line will only show up a few options. A better option to run this on Windows is to start "Vulkan Configurator" from the start menu.